gatewayconfig/bluetooth/characteristics/wifi_connect_characteristic.py

```diff
@@ -98,6 +114,16 @@ def WriteValue(self, value, options):
         self.wifi_service = str(wifi_details.service)
         self.wifi_password = str(wifi_details.password)
 
+        if self.wifi_service and self.wifi_password:
+            # Start connecting to the Wifi
+            self.connecting = True
+
+            try:
+                cmd_thread = CommandThread(self)
+                cmd_thread.start()
```

Comment:
when does the thread end? from code before this, it looks like unless an exception happens it doesn't exit. Does nmcli retrun non zero for an already connected connection.

Reply:
The thread will end after calling the nmcli command for connecting wifi. The status of the connection made is being checked alongside in the [timeout method](https://github.com/NebraLtd/hm-config/blob/posterzh/190/fix-wifi-connecting/gatewayconfig/bluetooth/characteristics/wifi_connect_characteristic.py#L87).

Reply:
yeah, my bad, I thought there was a infinite loop in there.

Reply:
that's fine.